catalog/core/catalog-core-versioning/versioning-common/src/main/java/ddf/catalog/core/versioning/MetacardVersion.java

```diff
@@ -209,31 +260,85 @@ public MetacardVersion(Metacard sourceMetacard, Action action, Subject subject)
         this.setTags(Collections.singleton(VERSION_TAG));
     }
 
+    private byte[] getVersionType(Metacard sourceMetacard) {
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+            oos.writeObject(sourceMetacard.getMetacardType());
+            return baos.toByteArray();
+        } catch (IOException e) {
+            throw new RuntimeException("Could not serialize MetacardType");
+        }
+    }
+
     /**
      * Returns a {@link BasicTypes#BASIC_METACARD} version of the given
      * {@link MetacardVersion}
      *
      * @return The converted metacard
      * @throws IllegalStateException
      */
-    public Metacard toBasicMetacard() {
-        return toBasicMetacard(this);
+    public Metacard getMetacard(List<MetacardType> types) {
+        return toMetacard(this, types);
     }
 
-    public static Metacard toBasicMetacard(Metacard source) {
+    public static Metacard toMetacard(Metacard source, List<MetacardType> types) {
         String id = (String) source.getAttribute(MetacardVersion.VERSION_OF_ID)
                 .getValue();
         if (isNullOrEmpty(id)) {
             throw new IllegalStateException(
                     "Cannot convert history metacard without the original metacard id");
         }
 
-        MetacardImpl result = new MetacardImpl(source, BasicTypes.BASIC_METACARD);
+        String typeString = (String) source.getAttribute(VERSION_TYPE)
+                .getValue();
+        Optional<MetacardType> typeFromExisting = types.stream()
+                .filter(mt -> mt.getName()
+                        .equals(typeString))
+                .findFirst();
+
+        MetacardType metacardType =
+                typeFromExisting.orElseGet(() -> getMetacardTypeBinary(source).orElseThrow(
+                        cannotDeserializeException));
+
+        MetacardImpl result = new MetacardImpl(source, metacardType);
         result.setId(id);
         result.setTags(getVersionTags(source));
+
+        sanitizeVersionAttributes(result);
         return result;
     }
 
+    private static void sanitizeVersionAttributes(/*Mutable*/ Metacard source) {
+        Consumer<String> nullifySourceAttribute = (s) -> source.setAttribute(new AttributeImpl(s,
+                (Serializable) null));
+        versionDescriptors.stream()
```

Comment:
Could but versionDescriptors has a public accessor, so Would like all the types it is inclusive of to be in the type (including basic since everything should have them)